### DISCUSSION TITLE
Container Proxy error

### DISCUSSION THREAD

**Discussion by hieunguyen8794** [engageub/InternetIncome#182] [General]
`Starting Earnapp container..
docker: Error response from daemon: Container 3ba65fc95aa46448a676a7666f511e5fe4b30029893093660f09573daf76a105 is restarting, wait until the container is running.
earnapp75e1f52f53337f7ee567f2250506957b1`

i get error when start earn app with proxy, i try switch TUN2PROXY but got same error. but same config, proxy still running on another server.

ps. log tun2proxy
`docker logs 79e0835d8e644102e13573f2ae1e4f8c72e9a11dc53a5304cb98ca10a33e010e
open: Operation not permitted
Cannot find device "tun0"
Cannot find device "tun0"
Cannot find device "tun0"
time="2024-12-15T06:47:26Z" level=info msg="[DIALER] set fwmark: 0x22b"
time="2024-12-15T06:47:26Z" level=fatal msg="[ENGINE] failed to start: create tun: operation not permitted"
open: Operation not permitted
Cannot find device "tun0"
Cannot find device "tun0"
Cannot find device "tun0"
time="2024-12-15T06:47:26Z" level=info msg="[DIALER] set fwmark: 0x22b"
time="2024-12-15T06:47:26Z" level=fatal msg="[ENGINE] failed to start: create tun: operation not permitted"
open: Operation not permitted
Cannot find device "tun0"
Cannot find device "tun0"
Cannot find device "tun0"`

**Comment by hieunguyen8794** [comment]
![image](https://github.com/user-attachments/assets/b8a4ec49-cd5d-46fc-b141-366f94e192ce)
pls help me check it

**Comment by engageub** [comment]
Hello,
Could you please run the following command and then start.

sudo bash internetIncome.sh --install

**Reply by hieunguyen8794** [reply]
Get:1 http://security.ubuntu.com/ubuntu focal-security InRelease [128 kB]
Hit:2 https://download.docker.com/linux/ubuntu focal InRelease                                                                           
Hit:3 http://us.archive.ubuntu.com/ubuntu focal InRelease                                
Get:4 http://us.archive.ubuntu.com/ubuntu focal-updates InRelease [128 kB]
Get:5 http://us.archive.ubuntu.com/ubuntu focal-backports InRelease [128 kB]
Fetched 383 kB in 5s (74.0 kB/s)  
Reading package lists... Done
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 docker.io : Depends: containerd (>= 1.2.6-0ubuntu1~)
E: Unable to correct problems, you have held broken packages.
Docker is installed.
Docker version 27.4.0, build bde2b89

i start again, and get same error

**Reply by engageub** [reply]
Well, you have to uninstall docker on your PC and install using the commands mentioned in the Readme File.
The output shows that you are using Docker version 27.4.0.
However, changes have been made in test branch to make it compatible with the new version.

**Comment by engageub** [comment]
Could you please confirm if the above command worked?
Which version of docker were you using. 
There are other possible solutions to add device. As the backward compatibility has not been tested, the changes were not pushed to main branch. 


Thank you

**Comment by engageub** [comment]
[Test branch](https://github.com/engageub/InternetIncome/tree/test) has been updated to use tun device instead of mapping from main host. 

The issue seems to appear after the update in containerd version v1.7.24 .
Please see https://github.com/containerd/containerd/issues/11078 for more details.




Thank you

**Reply by hieunguyen8794** [reply]
tks bro, i did it. Issue is fixed when use test branch
